packages/terminal/src/browser/base/terminal-widget.ts

```diff
@@ -24,6 +25,8 @@ export abstract class TerminalWidget extends BaseWidget {
 
     abstract processId: Promise<number>;
 
+    abstract processInfo?: Promise<TerminalProcessInfo>;
```

Comment:
you can do `:`, there won't be other implementations of `TerminalWidget`

at some point we will delete `TerminalWidget` interface and rename `TerminalWidgetImpl` to avoid bolierplate